🐛 [Bug]: EncryptCookie Middleware Crashes in FIPS-140 Mode

### Bug Description

When running an application that uses `middleware/encryptcookie`, a crash is observed during the cookie encryption process.  The bug seems to be triggered by [middleware/encryptcookie/utils.go:25](https://github.com/gofiber/fiber/blob/e334f54bf4840d75a47b9111151d1ffd254bc54f/middleware/encryptcookie/utils.go#L25) in both of the latest v2 and v3 codebases.

Current Code:
```
	block, err := aes.NewCipher(keyDecoded)
	if err != nil {
		return "", fmt.Errorf("failed to create AES cipher: %w", err)
	}

	gcm, err := cipher.NewGCM(block) // <-- Causes panic in FIPS mode
	if err != nil {
		return "", fmt.Errorf("failed to create GCM mode: %w", err)
	}
```

FIPS140-compliant Code:
```

	block, err := aes.NewCipher(keyDecoded)
	if err != nil {
		return "", fmt.Errorf("failed to create AES cipher: %w", err)
	}

	gcm, err := cipher.NewGCMWithRandomNonce(block) // <-- Works
	if err != nil {
		return "", fmt.Errorf("failed to create GCM mode: %w", err)
	}
```

I've confirmed setting a custom Encryptor to a function that uses `NewGCMWithRandomNonce` works without crashing.

### How to Reproduce

1. Using a service with encryptcookie middleware, 
2. Set the environment variable `export GODEBUG=fips140=only`
3. Launch the web service
4. Try running through a web workflow that uses an encrypted cookie
5. Observe a crash in the logs similar to:

```
panic: failed to create GCM mode: crypto/cipher: use of GCM with arbitrary 
IVs is not allowed in FIPS 140-only mode, use NewGCMWithRandomNonce

goroutine 84 [running]:
(program-module).errorHandler(0xc007850308?, {0xb3d560?, 0xc00007cf20?})
	error_handler.go:19 +0x2c
github.com/gofiber/fiber/v2.(*App).ErrorHandler(0xc0001a4508, 0xc007850308, 
{0xb3d560, 0xc00007cf20})
	.go/pkg/mod/github.com/gofiber/fiber/v2@v2.52.10/app.go:1061 +0x1e2
github.com/gofiber/fiber/v2.(*App).handler(0xc0001a4508, 0x4ad0d3?)
	.go/pkg/mod/github.com/gofiber/fiber/v2@v2.52.10/router.go:172 +0x88
github.com/valyala/fasthttp.(*Server).serveConn(0xc0001a6488, {0xb44950, 0xc0000d9888})
	.go/pkg/mod/github.com/valyala/fasthttp@v1.68.0/server.go:2465 +0x124b
github.com/valyala/fasthttp.(*workerPool).workerFunc(0xc0003a6000, 0xc000260400)
	.go/pkg/mod/github.com/valyala/fasthttp@v1.68.0/workerpool.go:225 +0x92
github.com/valyala/fasthttp.(*workerPool).getCh.func1()
	.go/pkg/mod/github.com/valyala/fasthttp@v1.68.0/workerpool.go:197 +0x32
created by github.com/valyala/fasthttp.(*workerPool).getCh in goroutine 181
	.go/pkg/mod/github.com/valyala/fasthttp@v1.68.0/workerpool.go:196 +0x194

```

### Expected Behavior

Cookie should be encrypted without crashing.

### Fiber Version

v2.52.10

### Code Snippet (optional)

```go

```

### Checklist:

- [x] I agree to follow Fiber's [Code of Conduct](https://github.com/gofiber/fiber/blob/main/.github/CODE_OF_CONDUCT.md).
- [x] I have checked for existing issues that describe my problem prior to opening this one.
- [x] I understand that improperly formatted bug reports may be closed without explanation.

--- a/middleware/encryptcookie/utils.go
+++ b/middleware/encryptcookie/utils.go
@@ -80,7 +80,7 @@ func DecryptCookie(name, value, key string) (string, error) {
 		return "", fmt.Errorf("failed to create GCM mode: %w", err)
 	}
 
-	if len(enc) < gcm.Overhead() {
+	if len(enc) < gcm.NonceSize()+gcm.Overhead() {
 		return "", ErrInvalidEncryptedValue
 	}
 

